### ISSUE TITLE
Ensure our dependencies are up-to-date

### ISSUE THREAD

**Issue by DanielMSchmidt** [wix/detox#680]
#### Description
I would like us to keep our tools sharp, therefore I would suggest to add a tool like [Renovate](https://renovateapp.com/) to help us keep our dependencies and dev dependencies up-to-date.

What it does is scanning this repository for the package.json files and updating the dependencies for them in individual PRs.
 
It provides a lot of configuration options, e.g. grouping certain dependencies together, automerging if the tests are green, etc.

**Comment by LeoNatan** [comment]
What JS dependencies does Detox have that would benefit from such a system?
The most important ones are in native and those often require work to merge.

**Comment by noomorph** [comment]
@DanielMSchmidt , maybe I'm a bit outdated on this subject and don't understand some latest things, but how do you think: maybe simple adding  `david-dm` badges already could make 80% of helpful impact for us? If yes, maybe you could prepare PR with this, of course if the repo owners will also agree to add it.

**Comment by noomorph** [comment]
@LeoNatan , the positive impact that we keep track of vulnerabilities in packages we use. Honestly speaking, I don't know whether we need to worry about vulnerabilities as detox is just a tool on a dev and CI environment that is not running 24/7 publicly open. But it is a nice to have indeed.

We can recall an example with `ws` package. It is listed here:

https://david-dm.org/wix/detox?path=detox

**Comment by DanielMSchmidt** [comment]
@noomorph I disagree there, the tool you linked only picks up one `package.json` it seems and I think having something telling us that we have outdated dependencies is not as helpful as something doing PRs with dependency updates.

**Comment by LeoNatan** [comment]
Since Detox is a local tool meant for development, I see vuln considerations as moot. On the other hand, we should strive to push versions of open source where it improves our development experience (such as Node).
